backend/app/util.py

```diff
@@ -26,10 +29,18 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-# FIXME: Idk how to return proper HTTPException's to the frontend
 def decode_jwt(encoded_jwt: str):
-    idinfo = id_token.verify_oauth2_token(
-        encoded_jwt, requests.Request(), get_settings().google_client_id
-    )
-
-    return GoogleAuth(**idinfo)
+    try:
+        idinfo = id_token.verify_oauth2_token(
+            encoded_jwt, requests.Request(), get_settings().google_client_id
+        )
+        return GoogleAuth(**idinfo)
+
+    # Error for malformed jwt's
+    except MalformedError:
+        raise HTTPException(status_code=498, detail="Could not parse JWT")
+
+    # Error for f.x expiration
+    except InvalidValue as e:
+        # Here we would refresh the token
+        raise HTTPException(status_code=498, detail=str(e))
```

Comment:
Does the error show up weirdly without the `str()`?

Reply:
I get an error without it

Reply:
Ah, okay